src/entity.rs

```diff
@@ -58,32 +62,18 @@ pub fn unescape(text: &str) -> Option<(String, usize)> {
     None
 }
 
-fn lookup(text: &str) -> Option<&'static str> {
-    let mut i = entity_data::ENTITIES.len() / 2;
-    let mut low = 0;
-    let mut high = entity_data::ENTITIES.len() - 1;
-
-    loop {
-        let cmp = text.cmp(entity_data::ENTITIES[i].0);
-        if cmp == Ordering::Equal {
-            return Some(entity_data::ENTITIES[i].1);
-        } else if cmp == Ordering::Less && i > low {
-            let mut j = i - ((i - low) / 2);
-            if j == i {
-                j -= 1;
-            }
-            high = i - 1;
-            i = j;
-        } else if cmp == Ordering::Greater && i < high {
-            let mut j = i + ((high - i) / 2);
-            if j == i {
-                j += 1;
-            }
-            low = i + 1;
-            i = j;
-        } else {
-            return None;
+fn lookup(text: &str) -> Option<&str> {
+    let entity_str = format!("&{};", text);
+
+    let entity = ENTITIES
+        .iter()
+        .find(|e| e.entity == entity_str);
```

Comment:
This lookup is O(_n_) in the size of the `ENTITIES` array. The old code did a binary search, O(log _n_), and I think we want to try to preserve that performance characteristic — otherwise a page full of `&zwnj;` will result in traversing all 2,200 members of the array for each lookup.

One way to do that is to do as  [`entities`' README](https://github.com/p-jackson/entities#readme) suggests and create a `HashMap` from the `ENTITIES` data, e.g. in a `lazy_static!` section. This would be a good first-level speedup. Possibly even faster would be to construct our own sorted array from the `ENTITIES` data such that we could still do the binary search, but the jury's out onto whether that or `HashMap` (or `BTreeMap`) would actually be faster.

(One step further would be to create the entity data lookup array at _build time_ with a `build.rs`, which would get us back to where we started, albeit sans Python.)